docs/user-guide/rules/performance-budget.md

```diff
@@ -0,0 +1,147 @@
+
+## What does the rule check?
+
+This rule calculates how long it will take to download all the resources loaded
+initially by the website under a `3G Fast` network. If the load time is
+**greater than 5 seconds**, the rule will fail.
+
+To calculate the final load time, some assumptions and simplifications are
+done. While the real numbers might be different, the results should provide
+enough guidance to know if something needs more attention.
+
+The reason for using predefined conditions and assumptions are:
+
+* Guarantee consistent results accross runs. If a website serves the same
+  assets, the results should be the same.
+* Show the impact in load time of each transmitted byte with the goal of
+  reducing the number and size of resources downloaded (first and third party).
+
+The simplified formula to calculate the time is:
+
+```text
+Time = (total number of requests * RTT) +
+       (number of different domains * RTT) +
+       (number of different secured domains * RTT) +
+       (number of redirects * RTT) +
+       (total number of requests * TCP slow-start phase) +
+       (total size of resources / bandwidth)
+```
+
+This is the list of things taken into account:
+
+* Everything is a first load, no values are cached, and no connections are
+  opened.
+* [`RTT` (Round-Trip Time)][rtt] is fixed and changes depending on the
+  configured network (more about this in #). It assumes all servers respond
+  instantly and in the same amount of time.
+* [`DNS lookup`][dns lookup]: Every hostname resolution requires 1 RTT,
+  imposing latency on the request and blocking the request while the lookup is
+  in progress.
+* [`TCP handshake`][three-way handshake]: Each request requires a new TCP
+  connection. TCP connections require 1 RTT before start sending information to
+  the server. There's no connection reuse and the maximum number of connections
+  to a domain (usually 6) is ignored.
+* [`TCP slow-start phase`][slow-start phase]: The values used to calculate the
+  duration are:
+  * `cwnd`: 10 network segments
+  * `rwnd`: 65,535 bytes (no `TCP window scaling`)
+  * `segment size`: 1460 bytes
+  After this phase, the full bandwidth of the connection is used to download
+  the remaining.
+* [`TLS handshake`][tls handshake]: New TLS connections usually require two
+  roundtrips for a "full handshake". However, there are ways of requiring only
+  1 RTT like [`TLS False Start`][tls false start] and [`TLS Session
+  Resumption`][tls session resumption]. This rule assumes the optimistic
+  scenario.
+* Redirects: We simplify the redirects and assume the connection is reused and
+  to the same server. If the redirect is to another domain, the penalty will be
+  even greater.
+* The server doesn't use [`HTTP2`][http2].
+
+## Can the rule be configured?
+
+You can change the type of connection and/or the target load time as follows:
+
+```json
+{
+    "performance-budget": ["error", {
+        "connectionType": "Dial",
+        "loadTime": 10
```

Comment:
Maybe document what happens if `loadTime` is not specified.

Reply:
So, restricting the the today size will be part of another rule?

Reply:
> restricting the the total size will be part of another rule?

I'm not sure about that. All things are connected (size, load time, network type, number of requests, etc.). I'd rather make this rule more intelligent (e.g.: take into account JS parse time) than creating a new one that just checks the size without any other variable (like connection speed).